mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -286,6 +286,26 @@ public String createExperiment(String experimentName) {
     return mapper.toCreateExperimentResponse(ojson).getExperimentId();
   }
 
+  /**
+   * Create a new experiment. This method allows providing all possible
+   * fields of CreateExperiment, and can be invoked as follows:
+   *
+   *   <pre>
+   *   import org.mlflow.api.proto.Service.CreateExperiment;
+   *   CreateExperiment.Builder request = CreateExperiment.newBuilder();
+   *   request.setName(name);
```

Comment:
shall we add `setTags` to the example here too?

Reply:
Can we add tests for this API?

Reply:
Done - weird proto-based syntax derived from https://stackoverflow.com/a/66740363/13837474

Reply:
Done